src/main/java/org/cascadebot/cascadebot/utils/votes/VoteButtonGroup.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2019 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.utils.votes;
+
+import net.dv8tion.jda.core.entities.Message;
+import net.dv8tion.jda.core.entities.User;
+import org.cascadebot.cascadebot.CascadeBot;
+import org.cascadebot.cascadebot.utils.buttons.ButtonGroup;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class VoteButtonGroup extends ButtonGroup {
+
+    private Map<Long, Object> votes = new HashMap<>();
+
+    private VoteButtonGroupBuilder.IVotePeriodicRunnable periodicRunnable;
+
+    private Timer timer = new Timer();
+
+    private Timer voreTimer;
+
+    public VoteButtonGroup(long ownerId, long channelId, long guildId, VoteButtonGroupBuilder.IVotePeriodicRunnable votePeriodicRunnable, Timer voteTimer) {
+        super(ownerId, channelId, guildId);
+        this.periodicRunnable = votePeriodicRunnable;
+        if(periodicRunnable != null) {
+            setUpVoteProcessConsumer();
+        }
+        this.voreTimer = voteTimer;
+    }
+
+    private void setUpVoteProcessConsumer() {
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                CascadeBot.INS.getShardManager().getGuildById(getGuildId()).getTextChannelById(getChannelId()).getMessageById(getMessageId()).queue(message -> {
+                    periodicRunnable.run(getOrderedVoteResults(), message);
+                });
+            }
+        }, 5000, 5000);
```

Comment:
This should be configurable

Reply:
Also this never seems to end?

Reply:
It does. i run timer.cancel in a different method.

Reply:
Okay, still needs to be configurable. 5 seconds might be over whatever the vote time is set to